runtime/tools/spec-builder/src/system-module/index.js

```diff
@@ -0,0 +1,34 @@
+const { stringToHex } = require('@polkadot/util');
+
+
+class System{
+
+    changesTrieConfig = null;
+    code = "";
+    static CODE = ":code";
+    constructor(changesTrieConfig, code){
+        this.changesTrieConfig = changesTrieConfig;
+        this.code = code;
+    }
+
+    /**
+     * Converts the system instance to raw object
+     * @param system instance of the class 
+     */
+    static toRaw(system) {
+        if (!system.code) {
+            throw new Error("Code property is not populated");
+        }
+        const systemRaw = Object.entries(system).map(pair => {
```

Comment:
- I am not sure why you are using the `map` instead of returning simply object with `key/value` pairs. You can easily do `Object.assign()` afterwards, in order to merge all key-value pairs.

Is there a reason for you to process the other keys (different than `code`) by doing `[stringToHex(pair[0]), stringToHex(pair[1])`. I think that you need to process only the `code` key. The rest of the you should be ignored as it is not supported (whatever it is)